### PR TITLE
Ensure headerCount is positive before making headers slice

### LINKED ISSUE #457: Panic: makeslice: len out of range
**Describe the bug**
Some messages cause panic as below:

```
goroutine 965 [running]:
github.com/segmentio/kafka-go.(*messageSetReaderV2).readMessage(0xc00169e408, 0x7cb6, 0xc001c33af0, 0xc001c33ae0, 0x2, 0xf, 0x0, 0x1, 0x0, 0x0, ...)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/message.go:557 +0x399
github.com/segmentio/kafka-go.(*messageSetReader).readMessage(0xc00169e3f0, 0x7cb6, 0xc001c33af0, 0xc001c33ae0, 0xc001c33a84, 0xf, 0x0, 0x0, 0xc001766000, 0x0, ...)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/message.go:136 +0x75
github.com/segmentio/kafka-go.(*Batch).readMessage(0xc00064b000, 0xc001c33af0, 0xc001c33ae0, 0x2, 0x0, 0xcf3400, 0x100000001, 0xc001766000, 0xc001766000, 0xc001c33ae0)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/batch.go:222 +0x78
github.com/segmentio/kafka-go.(*Batch).ReadMessage(0xc00064b000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/batch.go:178 +0xf2
github.com/segmentio/kafka-go.(*reader).read(0xc001c33ee8, 0x9c7ea0, 0xc00032c140, 0x7cb6, 0xc001814000, 0x0, 0x0, 0x0)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/reader.go:1372 +0x3cb
github.com/segmentio/kafka-go.(*reader).run(0xc001c33ee8, 0x9c7ea0, 0xc00032c140, 0x0)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/reader.go:1191 +0x2c2
github.com/segmentio/kafka-go.(*Reader).start.func1(0xc001d541a0, 0x9c7ea0, 0xc00032c140, 0x0, 0xfffffffffffffffe, 0xc001d542c0)
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/reader.go:1093 +0x1c6
created by github.com/segmentio/kafka-go.(*Reader).start
        /home/akamenskiy/go/pkg/mod/github.com/segmentio/kafka-go@v0.3.6/reader.go:1073 +0x18b
```

**Kafka Version**
2.4.1

**To Reproduce**
No idea how to reproduce.

**Expected behavior**
be able to fetch message

**Additional context**
compact topic, messages have no headers at all (headers not set) produced by Java application using official Kafka client libs (same version as kafka cluster).

### PR DESCRIPTION
**Pull request by godwhoa**
Other libraries seems to be doing:
https://github.com/Shopify/sarama/blob/8c3fe6939f5667b9390af8865f834995182b8b58/record.go#L104
https://github.com/apache/kafka/blob/927edfece3db8aab7d01850955f9a65e5c110da5/clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java#L356
https://github.com/dpkp/kafka-python/blob/a27ab881726ed1a2d952867a1fa266573165d6aa/kafka/record/default_records.py#L241

Might fix #457, not sure if it was due to an actual corrupted record or a bug elsewhere in the decoder.

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -551,6 +551,10 @@ func (r *messageSetReaderV2) readMessage(min int64,
 
 	headers = make([]Header, headerCount)
 
+	if headerCount >= 0 {
+		headers = make([]Header, headerCount)
+	}
```
Comment (achille-roussel) [inline message.go:556]: Do we need to remove the `headers = make([]Header, headerCount)` line above?

**Review by achille-roussel — COMMENTED** [review]
Thanks for the contribution @godwhoa!

File: message.go (hunk 1 above)
Comment (godwhoa) [reply message.go:556]: Ah, oops! Yes

**Comment by stankud** [comment]
Was this every merged into the master branch? Currently running into this issue in v0.4.16

**Pull request merged**